py4DSTEM/process/calibration/ellipse.py

```diff
@@ -602,4 +604,99 @@ def compare_double_sided_gaussian(data, p, power=1, mask=None):
 
     return
 
+## Create look up table for background subtraction
+def get_1Dbackground(data, qx0, qy0 ,e, phi, 
+                    mask_update_iter=3, 
+                    min_relative_threshold = 4, 
+                    smoothing_window_size = None, 
+                    smoothing_poly_order = 4, 
+                    smoothing_log = True, 
+                    min_background_value=1E-3, 
+                    return_polararr=False):
+    """
+    Gets the median polar background for a diffraction pattern
+
+    Accepts: 
+    data      (ndarray) the data for which to find the polar eliptical background, usually a diffraction pattern
+    qx0,qy0   (numbers) the ellipse center; if the braggpeaks have been centered,
+              these should be zero
+    e         (number) the length ratio of semiminor/semimajor axes
+    phi       (number) tilt of the major axis with respect  to (qx, qy) axis, in radians 
+
+    Returns: 
+    background1D (ndarray) 1D polar elliptical background
+    r_bins       (ndarray) the elliptical semimajor axis radius associated with background1D
+    PolarData    (ndarray) **optional: returns the masked polar transform from which the 1D background is computed
+    """
+    # Compute Polar Transform
+    PolarData, rr, tt = cartesianDataAr_to_polarEllipticalDataAr(data,tuple([qx0,qy0, 1, e, phi]))
+   
+    # Crop polar data to maximum distance which contains information from original image
+    if (PolarData.mask.sum(axis = (0))==PolarData.shape[0]).any():
+            Maximal_Distance  = np.min(np.where(PolarData.mask.sum(axis = (0))==PolarData.shape[0]))
+            PolarData = PolarData[:,0:Maximal_Distance]
+            r_bins = rr[0,0:Maximal_Distance]
+    else:
+            r_bins = rr[0,:]
```

Comment:
move part of this functionality to cartesianDataAr_to_polarEllipticalDataAr...
- in cartesianDataAr_to_polarEllipticalDataAr, set `r_range` based on maximum corner distance
- in this function, do what you're doing, but march inwards from the highest `r` value down until you find a good column